packages/lib/string-utils.ts

```diff
@@ -251,8 +251,20 @@ export function surroundKeywords(keywords: KeywordType, text: string, prefix: st
 		})
 		.join('|');
 	regexString = `(${regexString})`;
-	const re = new RegExp(regexString, 'gi');
-	return text.replace(re, `${prefix}$1${suffix}`);
+	const keywordRegex = new RegExp(regexString, 'gi');
+
+	return text
+		.split(keywordRegex)
+		.map((textSegment) => {
+			return {
+				text: textSegment,
+				shouldHighlight: keywordRegex.test(textSegment),
+			};
+		})
+		.reduce((highlightedText, textPart) => {
+			const nextPart = textPart.shouldHighlight ? `${prefix}${htmlentities(textPart.text)}${suffix}` : htmlentities(textPart.text);
+			return highlightedText + nextPart;
+		}, '');
```

Comment:
```suggestion
		.map((textSegment) => {
			const encodedText = htmlentities(textSegment);
			if (keywordRegex.test(textSegment)) {
				return `${prefix}${encodedText}${suffix}`;
			} else {
				return encodedText;
			}
		}).join('');
```
Doing this without a `.map -> .reduce` might be more readable :).

Reply:
Of course, I will simplify this.